docs/products/kafka/howto/connect-java.rst

```diff
@@ -0,0 +1,178 @@
+                 props.put("ssl.keystore.location", "client.keystore.p12");
+                 props.put("ssl.keystore.password", "secret");
+                 props.put("ssl.key.password", "secret");
+                 props.put("key.serializer",
+                     "org.apache.kafka.common.serialization.StringSerializer");
+                 props.put("value.serializer",
+                     "org.apache.kafka.common.serialization.StringSerializer");
+                 KafkaProducer<String, String> producer = new KafkaProducer<>(props);
+                 producer.send(new ProducerRecord<String, String>("demo-topic", "test-message"));
+                 producer.close();
+             }
+         }
+
+Create and run a consumer that pulls and outputs the data from ``demo-topic``:
+
+.. code::
+
+         import java.io.IOException;
+         import java.util.Arrays;
+         import java.util.Properties;
+         import org.apache.kafka.clients.consumer.ConsumerRecord;
+         import org.apache.kafka.clients.consumer.ConsumerRecords;
+         import org.apache.kafka.clients.consumer.KafkaConsumer;
+
+         public class Consumer {
+             public static void main(String[] args) throws IOException {
+                 Properties props = new Properties();
+                 props.put("bootstrap.servers", "service-uri.aivencloud.com:port");
+                 props.put("security.protocol", "SSL");
+                 props.put("ssl.truststore.location", "client.truststore.jks");
+                 props.put("ssl.truststore.password", "secret");
+                 props.put("ssl.keystore.type", "PKCS12");
+                 props.put("ssl.keystore.location", "client.keystore.p12");
+                 props.put("ssl.keystore.password", "secret");
+                 props.put("ssl.key.password", "secret");
+                 props.put("group.id", "demo-group");
+                 props.put("key.deserializer",
+                   "org.apache.kafka.common.serialization.StringDeserializer");
+                 props.put("value.deserializer",
+                   "org.apache.kafka.common.serialization.StringDeserializer");
+                 KafkaConsumer<String, String> consumer = new KafkaConsumer<>(props);
+                 consumer.subscribe(Arrays.asList("demo-topic"));
+                 while (true) {
+                     ConsumerRecords<String, String> records = consumer.poll(1000);
+                     for (ConsumerRecord<String, String> record : records) {
+                         System.out.printf("offset = %d, key = %s, value = %s",
+                             record.offset(), record.key(), record.value());
+                     }
+                 }
+             }
+         }
+
+Relying on SASL-SSL authentication
+-----------------------------------
+
+If you want to use SASL authentication, turn on ``kafka_authentication_methods.sasl`` under *Advanced configuration* on the *Overview* page of your service.
```

Comment:
We should have an article to reference here https://developer.aiven.io/docs/products/kafka/howto/kafka-sasl-auth.html

Reply:
done